### PR TITLE
AEA-3122 Common dev environment

### PR DESCRIPTION
**Pull request by anthony-nhs**

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    "context": "..",
+    "args": {}
+  },
+  "mounts": [
+    "source=${env:HOME}${env:USERPROFILE}/.aws,target=/home/vscode/.aws,type=bind",
+    "source=${env:HOME}${env:USERPROFILE}/.ssh,target=/home/vscode/.ssh,type=bind"
+  ],
+  // Features to add to the dev container. More info: https://containers.dev/features.
+  "features": {
+    "ghcr.io/devcontainers/features/docker-in-docker:latest": {
```
Comment (anthony-nhs) [inline .devcontainer/devcontainer.json:17]: use other docker option

File: README.md [hunk 2]
```diff
@@ -1,17 +1,23 @@
 # Electronic Prescription Service API
-
 ![Build](https://github.com/NHSDigital/electronic-prescription-service-api/workflows/Build/badge.svg?branch=master)
```
Comment (originalphil) [inline README.md:2]: Duplicate of the below?

File: README.md [hunk 3]
```diff
@@ -54,52 +71,82 @@ brew install git     # if not already installed
...
 
 Various scripts and commands rely on environment variables being set. These are documented with the commands.
 
 :bulb: Consider using [direnv](https://direnv.net/) to manage your environment variables during development and maintaining your own `.envrc` file - the values of these variables will be specific to you and/or sensitive.
 
+The following are recommended to place in the .envrc file
+
+```
+export PACT_PROVIDER=nhsd-apim-eps
+export PACT_BROKER_BASIC_AUTH_PASSWORD=<SECRET>
```
Comment (originalphil) [inline README.md:114]: Any tips on where to obtain these <SECRET>s?

File: README.md [hunk 4]
```diff
@@ -111,52 +158,96 @@ make
...
-* `build` -- Outputs the FHIR R4 validated models and artifacts for the: specification, coordinator and apigee proxies into the corresponding `dist/` directories
-* `test` -- Performs quality checks including linting, licence checking of dependencies and unit/low level integration tests
-* `release` -- Pulls all the artifacts for the individual components together and arranges them in a format ready to deploy; used mainly by CI but useful to check the output matches expectations
+- `clean` -- Removes the output from the build and release commands
+- `build` -- Outputs the FHIR R4 validated models and artifacts for the: specification, coordinator and apigee proxies into the corresponding `dist/` directories
+- `test` -- Performs quality checks including linting, licence checking of dependencies and unit/low level integration tests
+- `release` -- Pulls all the artifacts for the individual components together and arranges them in a format ready to deploy; used mainly by CI but useful to check the output matches expectations
 
-#### Run commands
+#### Run targes
```
Comment (originalphil) [inline README.md:166]: Typo - 'targes'.

File: README.md [hunk 5]
```diff
@@ -174,39 +265,42 @@ The Apigee portal will not automatically pull examples from schemas, you must sp
...
 
-## Validator
+# Validator
 
 The FHIR Validator is fetched during CI for a specific released tag. To see the released tag currently being used you can review the `Download Validator` step [version](azure/azure-build-pipeline.yml)
 
-### Running the validator locally
-
+## Running the validator locally
 You can also run the validator locally by cloning the repo in the parent folder of this checked out repo
```
Comment (originalphil) [inline README.md:289]: Duplicate of below.

File: .devcontainer/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+# Install ASDF
+RUN git clone https://github.com/asdf-vm/asdf.git ~/.asdf --branch v0.11.2; \
+    echo '. $HOME/.asdf/asdf.sh' >> ~/.bashrc; \
+    echo '. $HOME/.asdf/completions/asdf.bash' >> ~/.bashrc;
+
+ENV PATH="$PATH:/home/vscode/.asdf/bin/"
+
+# Install ASDF plugins
+RUN asdf plugin add python; \
+    asdf plugin-add poetry https://github.com/asdf-community/asdf-poetry.git; \
```
[the remark this answers is not in the document]
Comment (anthony-nhs) [reply .devcontainer/Dockerfile:37]: it does the same thing, but will change it for consistency

File: README.md (hunk 2 above)
Comment (anthony-nhs) [reply README.md:2]: my bad resolving merge conflicts - will fix

File: README.md [hunk 7]
```diff
@@ -54,52 +71,82 @@ brew install git     # if not already installed
...
-sdk default java 11.0.17-zulu     # make default
+```
+
+### asdf setup
+
+You need to run the following to install the needed asdf packages
+
+```
+asdf plugin add python
+asdf plugin-add poetry https://github.com/asdf-community/asdf-poetry.git
```
[the remark this answers is not in the document]
Comment (anthony-nhs) [reply README.md:86]: see above - will fix

File: README.md (hunk 5 above)
Comment (anthony-nhs) [reply README.md:289]: fixed

File: README.md (hunk 3 above)
Comment (anthony-nhs) [reply README.md:114]: will pass them to you - dont want to post on public forum about them

**Pull request merged**